docs/docs/learn/shards/working-with-data.md

```diff
@@ -1,11 +1,236 @@
-<!-- TODO -->
 
-## Passthrough ##
-<!-- TODO -->
+Now that you have learned to code with Shards and mastered the program flow, let us round off this series of primers by taking a look at how data works in Shards.
 
-## Scope ##
-<!-- TODO -->
+## Scope
 
-## Converting Data Types ##
-<!-- TODO -->
\ No newline at end of file
+### Local and Global
+
+Scope determines the visibility of data at different points in your program. When data is assigned to variables, these variables will either have a **local** or **global** scope.
+
+- Local: The variable is only known within the Wire it originated from.
+
+- Global: The variable is known throughout the entire Mesh.
+
+In the example below, the Wire `retrieve-x` attempts to retrieve the value of `.x` defined in the Wire `define-x`. Note how it is able to retrieve the value of 1 even though it was defined in a separate Wire. This is due to how `.x` has been defined as a global variable, making its value available to all Wires on the Mesh.
+
+=== "Command"
+    ```{.clojure .annotate linenums="1"}
+    (defmesh main)
+
+    (defwire get-x
+      (Get .x :Default 0) (Log "x"))
+
+    (defwire define-x
+      1 >== .x) ;; (1)
+
+    (schedule main define-x)
+    (schedule main get-x)
+    (run main)
+    ```
+
+    1. `>==` is the alias for the [`Set`](../../../reference/shards/General/Set/) shard, with the parameter `Global` set to **true**. This makes `.x` a global variable.
+
+=== "Output"
+    ```{.clojure .annotate linenums="1"}
+    [get-x] x: 1
+    ```
+
+For the following example, `get-x` fails to retrieve the value of `.x` defined in `define-x` and returns the default value of 0. This is due to how `.x` was locally defined within the Wire `define-x` and cannot be accessed by other Wires separate from it.
+
+=== "Command"
+    ```{.clojure .annotate linenums="1"}
+    (defmesh main)
+
+    (defwire get-x
+      (Get .x :Default 0) (Log "x"))
+
+    (defwire define-x
+      1 >= .x) ;; (1)
+
+    (schedule main define-x)
+    (schedule main get-x)
+    (run main)
+    ```
+
+    1. `>=` is the alias for the [`Set`](../../../reference/shards/General/Set/) shard, with the parameter `Global` set to **false**. This makes `.x` a local variable.
+
+=== "Output"
+    ```{.clojure .annotate linenums="1"}
+    [get-x] x: 0
+    ```
+
+### Flow Methods
+
+If Wire Y is run from a separate Wire X using methods such as [`Do`](../../../reference/shards/General/Do), the variables on Wire X will be copied such that Y has access to its value at the moment it was called.
+
+However, this does not mean that Wire Y is in the same scope as X. Wire Y holds only a copy of the value - it does not have access to the actual variable.
+
+If a method such as [`Step`](../../../reference/shards/General/Step) is used instead, Wire Y would be scheduled on Wire X itself, giving it the same scope and access to X's actual variables.
+
+![Methods to control the Flow of Shards can affect scope.](assets/wires-scope.png)
+
```

Comment:
Would be good to also include the code representing this situation (on top of the image, or below it).